### DISCUSSION TITLE
aptos move test --move-2

### DISCUSSION THREAD

**Discussion by MoonShiesty** [aptos-labs/aptos-developer-discussions#451] [Questions]
### Discord user ID

moonshiesty

### Describe your question in detail.

i tried out the new move-2 `enum` features. i was able to compile the module with the latest aptos cli, but most of my tests fail with the following error:

### What error, if any, are you getting?

```
┌── test_ ──────
│ Test did not error as expected. Expected test to abort with code 7005 originating in the module 0000000000000000000000000000000000000000000000000000000000001004::vault but instead it gave a UNEXPECTED_VERIFIER_ERROR (code 2017) error with error message: "Unexpected verifier/deserialization error! This likely means there is code stored on chain that is unverifiable!
│ Error: VMError { major_status: LOOKUP_FAILED, sub_status: None, message: None, exec_state: None, location: Module(ModuleId { address: 0000000000000000000000000000000000000000000000000000000000001002, name: Identifier("oracle") }), indices: [(FunctionHandle, 38)], offsets: [] }". Error originating in the module 0000000000000000000000000000000000000000000000000000000000001002::oracle rooted here
└──────────────────
```

tried aptos move clean

**Comment by wrwg** [comment]
Is this a result of `aptos move test`? Which CLI are you using, and which frameworks? The LOOKUP_FAILED error looks to me a potential result of using a newer CLI against an older framework.

**Reply by MoonShiesty** [reply]
im using aptos-cli 4.2.2 with the frameworks based on rev="main"

**Reply by wrwg** [reply]
Any chance to create a repro or share the code some way? This is supposed to work, and is tested. The LOOKUP_FAILED error is supposedly unrelated to enums, it means the VM is trying to load a module and why it verifies dependencies, an expected dependency, like a function, wasn't found.
